components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -1733,6 +1733,17 @@ public static ApplicationManagementServiceClient getApplicationManagementService
         }
     }
 
+    /**
+     * Method used to create the file name of the wsdl to be stored in the registry
+     *
+     * @param provider   Name of the provider of the API
+     * @param apiName    Name of the API
+     * @param apiVersion API Version
+     * @return WSDL file name
+     */
+    public static String createWsdlFileName(String provider, String apiName, String apiVersion) {
+        return provider + "--" + apiName + apiVersion + ".wsdl";
```

Comment:
shouldn't we just remove the provider from here?

Reply:
The registry will store the wsdl file with the provider name appended. Hence it cannot be removed from this method.